proj_eclipse/src/main/java/smartmag/models/BoxModel.java

```diff
@@ -83,14 +86,25 @@ protected BoxRecord getRecord() {
 		return record;
 	}
 
+	public BoxModel CreateBox(Box b)
+			throws SQLIntegrityConstraintViolationException {
+		BoxModel bm = getBoxModel(b);
+		if (bm == null)
+			bm = new BoxModel(b);
```

Comment:
`getBoxModel` non restituisce mai null: se non esiste lo crea lui
```suggestion
		BoxModel bm = getBoxModel(b);
```